stormpath/express-stormpath
# Add produces option for /login

Implements the produces option for /login according to [the framework spec](https://github.com/stormpath/stormpath-framework-spec/blob/master/login.md#feature-description).
#### How to verify
1. Checkout this branch.
2. Clone [express-stormpath-sample-project](https://github.com/stormpath/express-stormpath-sample-project).
3. Use `npm link` and `npm link express-stormpath` to link the library to the example project.
4. Start the server.
5. Navigate to `http://localhost:3000/login`.
6. Login with an existing user.
7. Open up the terminal and execute `$ http POST localhost:3000/login username=abc password=123 --json`.
8. Verify that you get a `400 Bad Request` with a `Content-Type: application/json; charset=utf-8` and `{"error":"Invalid username or password."}` body.
9. Redo step 7, but this time change the username and password to a real account.
10. Verify that you get a `200 OK` response with two `Set-Cookie` headers. One for setting the `access_token` cookie, and the other one for setting `refresh_token`.
11. Now modify your configuration, and change `web.produces` to `['text/html', '']` (hack because extender currently merges arrays... which needs to be fixed, empty string overwrites second array item.).
12. Verify that step 5-6 works, but that step 7 responds with a `404 Not Found` status.
13. Modify the `web.produces` config to `['application/json', '']`.
14. Verify that step 5 results in a `404 Not Found` and that steps 7-10 work as previous.
15. Modify the `web.produces` config again to `['', '']`.
16. Verify that both steps 5 and 7 result in a ´404 Not Found`.

Fixes #308.
